src/socket/socket.cr

```diff
@@ -273,6 +273,23 @@ class Socket < IO::FileDescriptor
       end
     end
   end
+
+  # Returns the hostname
+  #
+  # Maximum of 253 characters are allowed, with 2 bytes reserved for storage.
+  # In practice, many platforms will disallow anything longer than 63 characters.
+  # ```
+  # Socket.gethostname # => my.machine.name.local
```

Comment:
`Socket.gethostname # => "host.example.org"`

It returns a string and using an mDNS name for your stuff will probably break things, so not a good example IMO ;)